net/devif/devif_callback.c

```diff
@@ -267,7 +283,7 @@ FAR struct devif_callback_s *
             {
               /* No.. release the callback structure and fail */
 
-              devif_callback_free(NULL, NULL, list);
+              devif_callback_free(NULL, NULL, list_head, list_tail);
```

Comment:
fail to return the devif_callback_s?

Reply:
It looks like yes (I have not modified this existing part of code).

Reply:
Ok, let's fix it in the new patch

Reply:
Could you explain what's wrong with that part of code? As I understand, the code takes devif_callback_s structure slot from the free list (g_cbfreelist). Then it checks the network adapter status. If the network adapter does not exist anymore or if the network adapter is down, then the code returns devif_callback_s structure slot back into the free list (g_cbfreelist), and returns NULL signaling that devif_callback_alloc() operation can not be done.

Reply:
Yes, you are right, but the code is little bit complex. This PR simplify the error handling: https://github.com/apache/incubator-nuttx/pull/4575

Reply:
I've left comment in #4575.